[Docs] Add running Spark job on Databricks (#4161)
mlrun/mlrun: docs/concepts/functions-overview.md

* **local** - execute a Python or shell program 
* **{ref}`job <job-function>`** - run the code in a Kubernetes Pod
* **{ref}`dask <dask-overview>`** - run the code as a Dask Distributed job (over Kubernetes)

> * **{ref}`databricks <databricks>`** - run a Spark "job" on a Databricks cluster

Reviewer: run code on Databricks cluster (python scripts, Spark etc)

> * **{ref}`databricks <databricks>`** - run code on Databricks cluster (python scripts, Spark etc)

Reviewer 2: ```suggestion
* **{ref}`databricks <databricks>`** - Run code on a Databricks cluster as a Python script.
```

Author: @assaf758 - the current text is yours. Need you two to agree on this. Thanks

Reviewer 2: ok np
